### PR TITLE
Add GitHub Actions workflow for building release binaries

### PR DESCRIPTION
**Pull request by nightlark**
This adds a GitHub Actions workflow that builds Ninja release binaries for Windows, Linux, and macOS on each commit and stores them as artifacts. For published releases on GitHub, the compiled release binary archives are uploaded as assets.

Currently CMake is used instead of the Python script bootstrapping process, because it kept the steps more similar across platforms (using configure.py on Windows requires extra steps with vcvarsall.bat to make so that the MSVC compiler is found). If configure.py is preferred it should be fairly easy to switch.

Another issue that was mentioned is using the ubuntu-latest image (LTS 18.04), and wanting a binary that works with older versions of Linux. Two ideas to address this are static linking with system libraries, or switching the build to use a Docker image that has an older version of glibc and stdc++. These should also address issue #1533, which using Ubuntu 16.04 as the build image might not.

### CODE REVIEW

File: .github/workflows/release-ninja-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      uses: actions/upload-artifact@v1
+      with:
+        name: ninja-binary-archives
+        path: artifact
+
+    - name: Upload release asset
+      if: github.event.action == 'published'
+      uses: actions/upload-release-asset@v1.0.1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jhasse) [inline .github/workflows/release-ninja-binaries.yml]: Do I have to set this up somehow?

Comment (nightlark) [reply .github/workflows/release-ninja-binaries.yml]: The nice thing with GitHub Actions is that nothing needs to be done to set up the token (I think they create a token as needed with expiration time of a few hours).

Commits directly to the repository use a token that has read/write access. For security, PRs from forked repositories have a token with read-only access.

**Comment by nightlark** [comment]
I added a trigger for pull requests that should make it show up under checks for PRs from forked repositories.

Testing it with some forks I created, the checks seem to show up as expected in the PRs when adding a new workflow. I think it's most likely just a permissions thing, since the workflow is being added in a PR from a fork without collaborator/write access to this repository.

Some of the test PRs between forks:
https://github.com/nightlark/ninja/pull/3
https://github.com/ryan-mast/ninja/pull/1

**Comment by jhasse** [comment]
Ah I see.

Can you switch to Ubuntu 16.04? Better than nothing.

We also need to link Visual Studio's C++ runtime statically on Windows.

**Comment by nightlark** [comment]
Mm yea, I'll see if I can get those changes made within the next couple of days.

**Comment by jhasse** [comment]
I've merged this PR for now so that we can see the result on this repository in action.

(Thank you very much btw!)

**Comment by nightlark** [comment]
It seems to be working for forks, based on open PRs like #1674 that have had commits since the merge.

A quick update on creating Linux binaries -- I think I've figured out how to build a binary that will take care of issue #1533 (CentOS 7 GLIBCXX symbols not found), and that should work with glibc-based Linux distros released since around 2011.

**Comment by mathstuf** [comment]
Using CentOS 6 was our solution before we ended up needing other newer support (CentOS 7 is suitable for us since the older Debian/Ubuntu releases it "jumped" over aren't supported by that project either).

**Comment by jhasse** [comment]
Automatic uploading of release assets didn't work: https://github.com/ninja-build/ninja/releases/tag/v1.10.0
@nightlark Do you have an idea why?

**Comment by nightlark** [comment]
I’d guess that it was just a v1.10.0 tag that was created, rather than publishing a release like was done for v1.9.0? If you create/publish a release for the v1.10.0 tag it should trigger the build and upload the assets.
 
It should be fairly straightforward (though simpler when all the builds are in the same workflow file) to change the action to trigger on a `v*` tag, and create a release for uploading the assets to.

**Comment by jhasse** [comment]
I've created a release now: https://github.com/ninja-build/ninja/releases/tag/v1.10.0

Maybe I should have created the release before pushing the tag? Or created the tag and the release in one go with GitHub's UI?

**Comment by jhasse** [comment]
Ah it worked. I'm starting to understand.

> It should be fairly straightforward (though simpler when all the builds are in the same workflow file) to change the action to trigger when a v* tag gets added, and create a release for uploading the assets to.

Yes, that sounds like a good idea :)

**Pull request merged**